source/features/link-to-changelog.md.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import {FileIcon} from '@primer/octicons-react';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {buildRepoURL, getRepo} from '../github-helpers';
+
+const getCacheKey = (): string => `changelog:${getRepo()!.nameWithOwner}`;
+
+function parseFromDom(): string | false {
+	return select('.js-navigation-item [title="changelog.md" i]')?.textContent ?? false;
+}
+
+async function fetchFromApi(): Promise<string | false > {
+	const {repository} = await api.v4(`
+		repository() {
+			object(expression: "HEAD:") {
+				...on Tree {
+					entries {
+						name
+					}
+				}
+			}
+		}
+	`);
+
+	return repository.object.entries.find((file: {name: string}) => /changelog\.md/i.exec(file.name))?.name ?? false;
+}
+
+const doesChangelogExist = cache.function(async () => pageDetect.isRepoHome() ? parseFromDom() : fetchFromApi(), {
+	cacheKey: getCacheKey
+});
+
+async function init(): Promise<void | false> {
+	// Always prefer the information in the DOM
+	if (pageDetect.isRepoHome()) {
+		await cache.delete(getCacheKey());
+		void doesChangelogExist();
+		return;
+	}
+
+	const changelog = await doesChangelogExist();
+	if (!changelog) {
+		return false;
+	}
+
+	const url = buildRepoURL('blob', 'HEAD', changelog);
+	select('.details-reset.Details-element')!.before(
+		<a className="btn btn-sm btn-invisible mt-2 p-0" href={url} type="button">
+			<FileIcon/>
+			<span>View {changelog}</span>
+		</a>
+	);
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isSingleTag
```

Comment:
I was expecting this to be on the main Releases page, not (just) on the single tag. Changelog.md  includes all the versions, not just one.

Reply:
Can you give me a visual of what your thinking.

Do you want to put this together with `tag-changelog` (forgot the name) feature?

Reply:
The example URL I used in the feature request was https://github.com/nodeca/js-yaml/releases, not https://github.com/nodeca/js-yaml/releases/tag/4.0.0 (but I think it can still appear in every `/releases/* page)

I'm expecting this link to appear above/outside the tags, 2 examples:

<img width="984" alt="Screen Shot 1" src="https://user-images.githubusercontent.com/1402241/107402336-7511be80-6ac9-11eb-9b93-23920ec86584.png">

Reply:
> The example URL I used in the feature request was [nodeca/js-yaml/releases](https://github.com/nodeca/js-yaml/releases), not [nodeca/js-yaml@`4.0.0` (release)](https://github.com/nodeca/js-yaml/releases/tag/4.0.0) (but I think it can still appear in every `/releases/*` page)

Got you. I was dealing with the _pain point_ that when going to a release you can't see the changelog.



> I'm expecting this link to appear above/outside the tags, 2 examples (I'm not super happy about either one):

Let me give it a shot and see what I can come up with.